### PR TITLE
fix: Added a periodic ping mechanism

### PR DESCRIPTION
**Pull request by JigarJoshi**
- When user is heavy on pub-sub usecase and in case if there is no message exchange for long - underlying http connection will get timed out. This is to keep it alive by making periodic ping to server.

 - updated api submodule.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -25,6 +25,9 @@ type Driver struct {
 	Token        string      `json:"token,omitempty"`
 	URL          string      `json:"url,omitempty"`
 	Protocol     string      `json:"protocol,omitempty"`
+
+	DisablePing    bool `json:"disable_ping,omitempty"`
+	PingIntervalMs int  `json:"ping_interval_ms,omitempty"`
```
Comment (efirs) [inline config/config.go:30]: Can we change it to `PingInterval time.Duration`?

File: driver/driver.go [hunk 2]
```diff
@@ -457,5 +463,36 @@ func NewDriver(ctx context.Context, cfg *config.Driver) (Driver, error) {
 		return nil, err
 	}
 
-	return &driver{driverWithOptions: drv}, nil
+	var wg sync.WaitGroup
```
Comment (efirs) [inline driver/driver.go:466]: Can we move it to separate function?
Something like `wd, ch := startHealthPingLoop(cfg.PingInterval, drv)`

File: driver/driver.go [hunk 3]
```diff
@@ -457,5 +463,36 @@ func NewDriver(ctx context.Context, cfg *config.Driver) (Driver, error) {
...
+		t := time.NewTicker(time.Duration(intervalMs) * time.Millisecond)
+		defer t.Stop()
+
+		for {
+			select {
+			case <-t.C:
+			case <-ch:
+				return
+			}
+			_, _ = drv.Health(context.TODO())
```
Comment (efirs) [inline driver/driver.go:487]: Create context with timeout:
```
ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
_, _ = drv.Health(ctx)
cancel()
```

File: tigris/client.go [hunk 4]
```diff
@@ -44,7 +44,6 @@ func NewClient(ctx context.Context, cfg ...*config.Client) (*Client, error) {
 	if err != nil {
 		return nil, err
 	}
-
```
Comment (efirs) [inline tigris/client.go:47]: Is this accidental?

**Review by efirs — APPROVED** [review]

File: config/config.go (hunk 1 above)
Comment (JigarJoshi) [reply config/config.go]: changed.

File: driver/driver.go (hunk 2 above)
Comment (JigarJoshi) [reply driver/driver.go:471]: fixed.

**Review by efirs — APPROVED** [review]

**Pull request merged**